### ISSUE TITLE
Bad x-amz-content-sha256 on large upload

### ISSUE THREAD

**Issue by nestarz** [bradenmacdonald/deno-s3-lite-client#12]
Hello ! Thanks for your work !!

Do you have any idea why on large files (> 100mb) I got this issue (I tried multiple partSize, and I have set the size field to the exact content length of the file uploaded), also I am streaming the file using ReadableStream (works fine on small files).

```
error: Uncaught (in promise) Error: The provided &apos;x-amz-content-sha256&apos; header does not match what was computed.
    return new ServerError(response.status, code, message, { key, bucketName, resource, region });
           ^
    at Module.parseServerError (https://raw.githubusercontent.com/nestarz/deno-s3-lite-client/patch-1/errors.ts:85:12)
    at eventLoopTick (ext:core/01_core.js:165:11)
    at async Client.makeRequest (https://raw.githubusercontent.com/nestarz/deno-s3-lite-client/patch-1/client.ts:281:23)
```

On a side note, I also have issues with filenames that contains space or characters like `()`, if I sluggify them it works well.

**Comment by bradenmacdonald** [comment]
I'm not sure off the top of my head; I'll try to take a look when I get some time.

**Comment by nestarz** [comment]
I will setup a reproductible example if I succeed so we can work on it

**Comment by nestarz** [comment]
Here is a S3 reproductible example for special filenames that doesnt work, I'm trying to do the same with large files.

```
import "https://deno.land/std@0.184.0/dotenv/load.ts";
import { S3Client } from "https://raw.githubusercontent.com/nestarz/deno-s3-lite-client/patch-1/mod.ts";

const s3 = new S3Client({
  accessKey: Deno.env.get("AWS_ACCESS_KEY_ID")!,
  secretKey: Deno.env.get("AWS_SECRET_ACCESS_KEY")!,
  endPoint: Deno.env.get("S3_ENDPOINT_URL")!,
  region: Deno.env.get("AWS_BUCKET_REGION")!,
  bucket: Deno.env.get("DEFAULT_BUCKET")!,
  useSSL: true,
  pathStyle: true,
});

const key = "remixicon (1).woff2";
const size = 1 * 64 * 1024 * 1024 + 100;
const value = new Blob([
  new Uint8Array(size).map(() => Math.floor(255 * Math.random())),
]).stream();

s3.putObject(key, value, {
  size,
  partSize: 64 * 1024 * 1024,
  metadata: { "Content-Type": "application/octet-stream" },
});
```

**Comment by nestarz** [comment]
I made two PR that should resolve the issue, but need review before approval. The payload sha256 fix is a hack and you may find the original issue based on this ?

**Comment by bradenmacdonald** [comment]
Hey @nestarz Thanks again for reporting these issues and for the PRs which helped me. I have merged #15 and #16 which should fix both issues, and include new test cases so we don't have the problem in the future.
